### PR TITLE
Show token owner in options page

### PR DESCRIPTION
**Pull request by its-miroma**
Related to <https://github.com/refined-github/refined-github/issues/6000#issuecomment-1556136109>

## Screenshot

why_is_that_token_not_hidden

### CODE REVIEW

**Review by fregante — COMMENTED** [review]
Thank you! Can you move it next to the field instead? Where this appears:

Screenshot 2024-06-02 at 14 16 03

Mockup:

Screenshot 2024-06-02 at 14 19 17

**Comment by its-miroma** [comment]
ٍSome longer usernames and names may not fit next to the token input box.

**Comment by fregante** [comment]
I see space for about 35 characters, if someone has a username that long they're seeing it wrapping/clipping/causing issues all over the interface already.
Screenshot 2024-06-02 at 14 34 59

and even then, it's already very close to the limit:

Screenshot 2024-06-02 at 14 36 44

**Comment by its-miroma** [comment]
No, I mean that I had implemented `show-names`, which would show `@its-miroma (Miroma)`

**Comment by fregante** [comment]
Ah yes, that part should be removed. `show-names` doesn't show your own username either for example.

File: source/options.tsx [hunk 1]
```diff
@@ -114,9 +114,12 @@ async function validateToken(): Promise<void> {
 	reportStatus({text: 'Validating…', tokenType});
 
 	try {
+		const scopes = await getTokenScopes(tokenField.value);
+		const text = `👤 @${await getNameFromToken(tokenField.value)}`;
```
Comment (fregante) [inline source/options.tsx:118]: Better parallelize them:
```suggestion
		const [scopes, user] = await Promise.all([
			getTokenScopes(tokenField.value),
			getNameFromToken(tokenField.value),
		]);
```

The emoji is part of the UI so it should be added near the DOM/HTML generation

Comment (its-miroma) [reply source/options.tsx]: > The emoji is part of the UI so it should be added near the DOM/HTML generation

I'm sorry, I'm unsure how to do that, I can't find any place where the HTML is generated. The images seem to get added through CSS

**Comment by fregante** [comment]
I like it!

Untitled

Thank you Miroma!

**Comment by fregante** [comment]
I forgot to mention this: it lacks support for GitHub Enterprise and it will break for GHE users. Can you add it? It looks like this:

https://github.com/refined-github/refined-github/blob/eb1d1af666dfd5de3cf783728b47dfd22fa0e746/source/options.tsx#L62-L66

**Pull request merged**